protobuf-lite/src/main/java/io/grpc/protobuf/lite/ProtoLiteUtils.java

```diff
@@ -132,6 +132,9 @@ public T parse(InputStream stream) {
               int position = 0;
               while ((chunkSize = stream.read(buf, position, size - position)) != -1) {
                 position += chunkSize;
+                if (chunkSize == 0 && position == size) {
```

Comment:
That seems equally fine to me as to my suggestion of using `if (position == size)` (without the additional `read`); your choice.

The additional `read()` was just to notice when an `InputStream` lied. While it would be nice, I can live without it. (The additional call to `read()` was already happening in the working case before; so I wasn't too concerned about the "useless" read.)